docs/developing-component.md

```diff
@@ -87,6 +87,23 @@ cp ./dist/darwin_amd64/debug/daprd ~/.dapr/bin
 go get -u github.com/dapr/components-contrib@master
 go mod tidy
 ```
-5. Import your component to Dapr [main.go](https://github.com/dapr/dapr/blob/d17e9243b308e830649b0bf3af5f6e84fd543baf/cmd/daprd/main.go#L79)
-6. Register your component in Dapr [main.go](https://github.com/dapr/dapr/blob/d17e9243b308e830649b0bf3af5f6e84fd543baf/cmd/daprd/main.go#L153-L226)
+5. Import your component to Dapr [main.go](https://github.com/dapr/dapr/blob/b3e1fe848de3ea7b297c712d188136919d314887/cmd/daprd/main.go#L20-L115)
+6. Register your component in Dapr [main.go](https://github.com/dapr/dapr/blob/b3e1fe848de3ea7b297c712d188136919d314887/cmd/daprd/main.go#L256-L385)
 7. Create a pullrequest in [Dapr](https://github.com/dapr/dapr/pulls)
+
+## Version 2 and beyond of a component
+
+API versioning of Dapr components follows the same approach as [Go modules](https://blog.golang.org/v2-go-modules) where the unstable version (v0) and first stable version (v1) are contained in the root directory of the component package.  For example v1 of the Redis binding component is located in `bindings/redis`. Code changes may continue in this package provided there are no breaking changes. Breaking changes include:
+
+* Renaming or removing a `metadata` field that the component is currently using
+* Adding a required `metadata` field or optional field that does not have a backward-compatible default value
+* Changing the component's logic such that it deviates from previously expected behavior
```

Comment:
This is very subjective. Can it be clearer? This can be interpreted as "no code change allowed".

Reply:
How about something like

```
Making significant changes to the component's behavior that would adversely affect existing users
```

Then it seems like the subjectivity could come from the word "adversely". But is it more clear? Open to suggestions.

Reply:
There will always be subjectivity. It is better now because not every change will cause that. People can still abuse this statement but maintainers exist for a reason.